check-submodules.ps1

```diff
@@ -0,0 +1,163 @@
+$answeredAll = $false   # has the user answered "yes to all" to init'ing submodules?
+$ProjectRoot = $PSScriptRoot
+$CleanAllSentinel = "$ProjectRoot\.cleansourcebuildsubmodules"
+
+# ask for confirmation and initialize a submodule if approved.
+function init_submodule($Path) {
+  $done = $false
+  while (-Not $done) {
+    Write-Warning "submodule $Path does not appear to be initialized."
+    Write-Host "Should I initialize it for you [Yes/no/all/quit]"
```

Comment:
Seems like the user might try to type the whole word and answer more prompts than they intended to. `[y]es / [n]o / [a]ll / [q]uit`?

Reply:
Changed as suggested.